app/settings/user-settings/user-settings.controller.js

```diff
@@ -34,29 +52,52 @@ function (
 
     UserSettingsEndpoint.getFresh({id: $rootScope.currentUser.userId}).$promise.then((settings) => {
         _.each(settings.results, (setting) => {
+            setting.user_id = setting.user.id;
+
             if (setting.config_key === 'hdx_api_key') {
-                $scope.hxlApiKey = '*** *** *** *** *** *** *** ' + setting.config_value.slice(setting.config_value.length - 4);
-                $scope.hxlApiId = setting.id;
+                setting.config_value = '*** *** *** *** *** *** *** ' + setting.config_value.slice(setting.config_value.length - 4);
+                $scope.hdxSettings.hdx_api_key = setting;
+                $scope.hxlApiKeySet = true;
+            }
+            if (setting.config_key === 'hdx_maintainer_id') {
+                $scope.hxlMaintainerSet = true;
+                $scope.hdxSettings.hdx_user_id = setting;
             }
         });
     });
 
-    function newKey() {
-        $scope.hxlApiKey = false;
+    function changeKey() {
+        $scope.hxlApiKeySet = false;
+    }
+
+    function changeId() {
+        $scope.hxlMaintainerSet = false;
     }
 
     function goToHdxView() {
         $state.go('settings.hdx');
     }
 
     function saveKey() {
-        $scope.api.$dirty = true;
-        if ($scope.apiKey) {
-            UserSettingsEndpoint.saveCache({id: $scope.hxlApiId,  user_id: $rootScope.currentUser.userId,  config_key: 'hdx_api_key', config_value: $scope.apiKey}).$promise.then((response) => {
-                $scope.hxlApiKey = '*** *** *** *** *** *** *** ' + $scope.apiKey.slice($scope.apiKey.length - 4);
-                $scope.hxlApiId = response.id;
-                Notify.notifyAction('settings.user_settings.api_key_saved', null, false, 'thumb-up', 'circle-icon confirmation', {callback: goToHdxView, text: 'settings.user_settings.start_tagging', callbackArg: null, actionClass: 'button button-alpha'});
-            });
+        var calls = [];
+
+        if ($scope.api['api-key'].$dirty) {
+            calls.push(
+                UserSettingsEndpoint.saveCache($scope.hdxSettings.hdx_api_key).$promise
+            );
+        }
+
+        if ($scope.api.hdx_maintainer_id.$dirty) {
+            calls.push(
+                UserSettingsEndpoint.saveCache($scope.hdxSettings.hdx_user_id).$promise
+            );
         }
+
+        $q.all(calls).then((response) => {
+            $scope.hxlMaintainerSet = true;
+            $scope.hxlApiKeySet = true;
+            $scope.hxlApiKey = '*** *** *** *** *** *** *** ' + $scope.hdxSettings.hdx_api_key.config_value.slice($scope.hdxSettings.hdx_api_key.config_value.length - 4);
```

Comment:
this will probably fail the lint validation

Reply:
which linter can I test this with?

Reply:
ehh. ignore me. It passes. 🤦‍♀️ IDK but it looked like way more than 100 chars to me : ( sorry